nvflare/apis/server_engine_spec.py

```diff
@@ -115,3 +115,15 @@ def persist_components(self, fl_ctx: FLContext, completed: bool):
 
     def restore_components(self, snapshot: RunSnapshot, fl_ctx: FLContext):
         pass
+
+    def send_admin_requests(self, requests):
+        pass
+
+    def check_client_resources(self, resource_reqs):
```

Comment:
we should add @abstractmethod decorator to these methods so that the child class is required to implement these methods.